app/src/main/java/org/hackillinois/androidapp2019/view/ScannerFragment.kt

```diff
@@ -1,219 +1,145 @@
-            false -> {
-                AlertDialog.Builder(context).apply {
-                    setMessage("User not registered, or already scanned in for event.")
-                    setPositiveButton("Ok") { dialog, _ ->
-                        shouldParseScan = true
-                        dialog.cancel()
-                    }
-                    create()
-                }.show()
-            }
-            true -> {
-                lastSuccessfullyScannedUser = lastScanStatus.userId
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
 
-                AlertDialog.Builder(context).apply {
-                    setMessage("Success: $lastSuccessfullyScannedUser")
-                    setPositiveButton("Ok") { dialog, _ ->
-                        shouldParseScan = true
-                        dialog.cancel()
-                    }
-                    create()
-                }.show()
-                staffOverrideSwitch.isChecked = false
-            }
-            null -> {
-                shouldParseScan = true
-                return
+        if (requestCode == PERMISSIONS_REQUEST_ACCESS_CAMERA) {
+            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                codeScanner.startPreview()
             }
         }
+    }
 
+    override fun onResume() {
+        super.onResume()
+        codeScanner.startPreview()
+    }
+
+    override fun onPause() {
+        codeScanner.releaseResources()
+        super.onPause()
     }
 
-    /**
-     * Extracts the userId from the HackIllinois QR String URI.
-     * Given an example URI "hackillinois://user?userid=github0000001"
-     * it is enough to grab everything after the last (and only) equals sign.
-     *
-     * If the URI becomes more complex, we can use actual URI decoders.
-     * @return the userId from a QR String
-     */
     private fun getUserIdFromQrString(qrString: String): String {
         val splitOnEquals = qrString.split("=")
         return splitOnEquals.last()
     }
 
-    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
-        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
-
-        if (requestCode == PERMISSIONS_REQUEST_ACCESS_CAMERA) {
-            if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-                qrScanner.decodeContinuous(onQrCodeScan)
+    private fun displayScanResult(lastScanStatus: ScanStatus?) = lastScanStatus?.let {
+        val message = when (it.lastScanWasSuccessful) {
+            true -> {
+                staffOverrideSwitch.isChecked = false
+                "Success: ${lastScanStatus.userId}"
             }
+            false -> "User not registered, or already scanned in for event."
```

Comment:
This could be a separate issue, but it would be nice to know the exact problem, i.e. not registered vs already scanned

Reply:
This! If there's one thing you guys do, it should be differentiating the various errors. What's great is that, with the latest form of errors, you get a very specific message (I think).